descwl_shear_sims/sim.py

```diff
@@ -174,30 +171,41 @@ def make_sim(
         survey_name=survey_name,
     ).pixel_scale
 
-    if (
-        hasattr(galaxy_catalog.layout, "wcs")
-        and hasattr(galaxy_catalog.layout, "bbox")
-        and not simple_coadd_bbox
-    ):
-        coadd_wcs = galaxy_catalog.layout.wcs
-        coadd_bbox = galaxy_catalog.layout.bbox
-    else:
-        if not isinstance(coadd_dim, int):
-            raise ValueError(
-                "coadd_dim should be int when galaxy catalog does not",
-                "have attribute 'layout'",
-            )
-        if simple_coadd_bbox:
-            coadd_wcs, coadd_bbox = make_coadd_dm_wcs_simple(
-                coadd_dim,
-                pixel_scale=pixel_scale,
+    if simple_coadd_bbox:
+        # Force to use a simple coadd boundary box
+        # where the center of the boundary box is the image origin and it
+        # matches to the world origin of the catalog's layout. Note that he
+        # center of the boundary box is the image_origin of the single exposure
+        # (SE).
+        if hasattr(galaxy_catalog.layout, "wcs"):
+            origin = galaxy_catalog.layout.wcs.getSkyOrigin()
+            world_origin = galsim.CelestialCoord(
+                ra=origin.getRa().asDegrees() * galsim.degrees,
+                dec=origin.getDec().asDegrees() * galsim.degrees,
             )
+        else:
+            world_origin = WORLD_ORIGIN
+        coadd_wcs, coadd_bbox = make_coadd_dm_wcs_simple(
+            coadd_dim,
+            pixel_scale=pixel_scale,
+            world_origin=world_origin,
+        )
+    else:
+        if (
+            hasattr(galaxy_catalog.layout, "wcs")
```

Comment:
I don't like the wcs "hiding" in the galaxy catalog.

Maybe we should have an optional argument to `make_sim` that can hold coadd information.

e.g.
```python
coadd_info = {'coadd_wcs': ..., 'coadd_bbox': ...}\
data = make_sim(..., coadd_info=coadd_info)
```

Reply:
Note in layout.py the wcs and bbox are not actually used.

Reply:
What about this?